tests/test_georaster.py

```diff
@@ -544,33 +544,83 @@ def test_reproject(self, example: str) -> None:
         # Reference raster to be used
         r = gr.Raster(example)
 
-        # Create an artificial raster with different bounds, shape and resolution
-        new_data = np.ones(r.shape)
-        transform = list(r.transform)
-        new_transform = rio.transform.Affine(
-            transform[0] / 2,
-            transform[1],
-            (r.bounds.left + r.bounds.right) / 2,
-            transform[3] / 2,
-            transform[4],
-            (r.bounds.bottom + r.bounds.top) / 2,
+        # -- Check proper errors are raised if nodata are not set -- #
+        r_ndv = r.copy()
+        r_ndv.nodata = None
+        default_ndv = _default_ndv(r_ndv.dtypes[0])
+        r_ndv.data.data[~r_ndv.data.mask][0] = default_ndv
+
+        # 1 - if no src_nodata is set and masked values exist, raises an error
+        if np.sum(r_ndv.data.mask) > 0:
+            with pytest.raises(ValueError, match="No nodata set, use `src_nodata`"):
+                r_ndv.reproject(dst_res=r_ndv.res[0] / 2, dst_nodata=0)
+
+        # 2 - if no dst_nodata is set and default value conflicts with existing value, a warning is raised
+        with pytest.warns(
+            UserWarning,
+            match="For reprojection, dst_nodata must be set. Default chosen value .* exist in self.data. \
+This may have unexpected consequences. Consider setting a different nodata with self.set_ndv.",
+        ):
+            r_ndv.reproject(dst_res=r_ndv.res[0] / 2, src_nodata=default_ndv)
+
+        if r.nodata is None:  # specific for the landsat test case, default nodata 255 cannot be used, so use 0
+            r.set_ndv(0)
+
+        # -- Additional tests -- #
+
+        # - Create 2 artificial rasters -
+        # for r2b, bounds are cropped to the upper left by an integer number of pixels (i.e. crop)
+        # for r2, resolution is also set to 2/3 the input res
+        min_size = min(r.shape)
+        rand_int = np.random.randint(min_size / 10, min(r.shape) - min_size / 10)
```

Comment:
We probably want to add `np.random.seed(42)` above that line, to have the tests fully consistent.

Reply:
The test should work regardless of the value of that number, if the range is set correctly. I would leave it purely random, unless it becomes an issue later.